deployer.go

```diff
@@ -121,9 +121,7 @@ func (d *deployer) deployUnit(wantedUnit *schema.Unit) error {
 
 	wuf := schema.MapSchemaUnitOptionsToUnitFile(wantedUnit.Options)
 	cuf := schema.MapSchemaUnitOptionsToUnitFile(currentUnit.Options)
-	wufUnescapedString := strings.Replace(wuf.String(), "\\\n", "", -1)
-	if strings.Replace(wufUnescapedString, " ", "", -1) != strings.Replace(cuf.String(), " ", "", -1) {
-
+	if wuf.Hash() != cuf.Hash() {
```

Comment:
But thats okay right? Fleet won't format it so it will notice a difference in whitespace once and then the hashes will match, right?

On 19 June 2015 13:33:41 BST, Catalin Ilea <notifications@github.com> wrote:
>> @@ -121,9 +121,7 @@ func (d *deployer) deployUnit(wantedUnit
>*schema.Unit) error {
>>  
>>  	wuf := schema.MapSchemaUnitOptionsToUnitFile(wantedUnit.Options)
>>  	cuf := schema.MapSchemaUnitOptionsToUnitFile(currentUnit.Options)
>> -	wufUnescapedString := strings.Replace(wuf.String(), "\\\n", "", -1)
>> -	if strings.Replace(wufUnescapedString, " ", "", -1) !=
>strings.Replace(cuf.String(), " ", "", -1) {
>> -
>> +	if wuf.Hash() != cuf.Hash() {
>
>func (u *UnitFile) Hash() Hash {
>	return Hash(sha1.Sum(u.Bytes()))
>}
>
>This won't work if there are extra " ", "\n", "\" in there as the hash
>will be different.
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Financial-Times/coco-fleet-deployer/pull/4/files#r32824384

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.